File: app/src/main/java/org/strickland/japa/InfoActivity.java
package org.strickland.japa;

import android.content.pm.PackageInfo;
import android.content.pm.PackageManager;
import android.content.res.AssetManager;
import android.os.Bundle;
import android.widget.ImageButton;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;

public class InfoActivity extends AppCompatActivity {

    private TextView tvInfoContent;
    private TextView appVersionName;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_info);

        tvInfoContent = findViewById(R.id.tv_info_content);
        appVersionName = findViewById(R.id.appVersionName);
        try {
            PackageInfo pInfo = getPackageManager().getPackageInfo(getPackageName(), 0);
            String versionName = pInfo.versionName;
            //int verCode = pInfo.versionCode;
            appVersionName.setText("v"+versionName);
        } catch (PackageManager.NameNotFoundException e) {
            e.printStackTrace();
        }
        String text = "nothing here.";
        AssetManager assetManager = getAssets();
        try (InputStream is = assetManager.open("info.txt");
             BufferedReader reader = new BufferedReader(new InputStreamReader(is))) {

            StringBuilder content = new StringBuilder();
            String line;
            while ((line = reader.readLine()) != null) {
                content.append(line).append("\n");
            }
            text = content.toString();
        } catch (IOException e) {
            e.printStackTrace();
        }
        tvInfoContent.setText(text);
        ImageButton btnClose = findViewById(R.id.btn_close_info);
        btnClose.setOnClickListener(v -> finish());
    }
}

File: app/src/main/kotlin/org/strickland/japa/InfoActivity.kt
package org.strickland.japa

import android.content.pm.PackageManager
import android.os.Bundle
import android.view.View
import android.widget.ImageButton
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import io.noties.markwon.Markwon
import io.noties.markwon.ext.tables.TablePlugin
import java.io.BufferedReader
import java.io.IOException
import java.io.InputStreamReader

class InfoActivity : AppCompatActivity() {
    //    private TextView tvInfoContent;
    //    private TextView appVersionName;
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_info)

        val tvInfoContent = findViewById<TextView>(R.id.tv_info_content)
        val appVersionName = findViewById<TextView>(R.id.appVersionName)
        try {
            val pInfo = getPackageManager().getPackageInfo(getPackageName(), 0)
            val versionName = pInfo.versionName
            //int verCode = pInfo.versionCode;
            appVersionName.setText("v" + versionName)
        } catch (e: PackageManager.NameNotFoundException) {
            e.printStackTrace()
        }
        var text = "nothing here."
        val assetManager = getAssets()
        try {
            assetManager.open("info.md").use { `is` ->
                BufferedReader(InputStreamReader(`is`)).use { reader ->
                    val content = StringBuilder()
                    var line: String?
                    while ((reader.readLine().also { line = it }) != null) {
                        content.append(line).append("\n")
                    }
                    text = content.toString()
                }
            }
        } catch (e: IOException) {
            e.printStackTrace()
        }


        //tvInfoContent.setText(text);

        //Markwon markwon = Markwon.create(this);
        val markwon = Markwon.builder(this)
            .usePlugin(TablePlugin.create(this))
            .build()
        markwon.setMarkdown(tvInfoContent, text)


        val btnClose = findViewById<ImageButton>(R.id.btn_close_info)
        btnClose.setOnClickListener(View.OnClickListener { v: View? -> finish() })
    }
}
